processing_options.go

```diff
@@ -652,19 +658,30 @@ func defaultProcessingOptions(acceptHeader string) (*processingOptions, error) {
 		UsedPresets: make([]string, 0, len(conf.Presets)),
 	}
 
-	if (conf.EnableWebpDetection || conf.EnforceWebp) && strings.Contains(acceptHeader, "image/webp") {
+	if (conf.EnableWebpDetection || conf.EnforceWebp) && strings.Contains(headers.Accept, "image/webp") {
 		po.Format = imageTypeWEBP
 	}
-
+	if len(headers.ViewportWidth) > 0 && conf.EnableClientHints {
+		po.Width, err = strconv.Atoi(headers.ViewportWidth)
+		if err != nil {
+			po.Width = 0
+		}
+	}
+	if len(headers.Width) > 0 && conf.EnableClientHints {
```

Comment:
Checking bool is faster, so let's check it first
```suggestion
	if conf.EnableClientHints && len(headers. Width) > 0 {
```